docs/source/serving/sky-serve.rst

```diff
@@ -521,3 +521,73 @@ The :code:`resources` field has the same spec as a normal SkyPilot job; see `her
+Speedup Setup in Model Serving
+------------------------------
+
+When serving AI models, the setup process like dependencies installation and model weights downloading takes a lot of time on the cold-start time. According to one experiment we found that, dependencies installation can take 45% of times (271s out of 608s) of the setup process, and 16% for model weights downloading (98s out of 608s). To speed up this process, you can use the following methods:
+
+Use a Custom Image for VMs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use a custom image that has the dependencies preinstalled (also probably model weights preloaded). This can be done by creating a VM, installing the dependencies, and then creating a custom image from the VM on the cloud console. Then, you can use the custom image to launch the VMs for serving:
+
+.. code-block:: yaml
+  :emphasize-lines: 7-8
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      ports: 8080
+      accelerators: A100:1
+      cloud: gcp
+      image_id: projects/my-project/global/machineImages/image-with-dependency-installed
+
+    # Here goes the setup and run commands...
+
+Notice that the :code:`cloud` field must be specified when :code:`image_id` is used. You can use :code:`any_of` in :code:`resources` to make it support multiple clouds:
+
+.. code-block:: yaml
+  :emphasize-lines: 7-13
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      ports: 8080
+      accelerators: A100:1
+      any_of:
+        - cloud: gcp
+          image_id: projects/my-project/global/machineImages/image-with-dependency-installed
+        - cloud: aws
+          image_id:
+            us-east-1: ami-0729d913a335efca7
+            us-west-2: ami-050814f384259894c
+
+    # Here goes the setup and run commands...
+
+Notice that AWS needs a per-region image since its image is only valid in a single region. This requires to configure a lot of images on the cloud console, but it can significantly speed up the dependency installation. It also takes the advantage of the in-region fast network, which delivers a quicker download speed of the machine image.
+
+Use Docker Container as Runtime Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also :ref:`use docker containers as runtime environment <docker-containers-as-runtime-environments>`. This can be done by creating a Docker container that has the dependencies preinstalled, and then using the container to launch the VMs for serving:
+
+.. code-block:: yaml
+  :emphasize-lines: 7
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      ports: 8080
+      accelerators: A100:1
+      image_id: docker:docker-image-with-dependency-installed
+
+    # Here goes the setup and run commands...
+
+This is easier to configure than machine images, but it may have a longer startup time than machine images since it needs to pull the docker image from the registry.
```

Comment:
```suggestion
This is easier to configure than machine images, but it may have a longer startup time than machine images since it needs to pull the docker image from the registry in addition to the usual VM setup.
```